### ISSUE TITLE
Possibility to hide fields in admin form based on roles of current user

### ISSUE THREAD

**Issue by jordygroote** [sulu/sulu#5606]
# Problem description

I had a situation where i would like to hide a content type from a page/article for everyone other than the administrator.

# Proposed solution

I think @nnatter proposed a useful feature, to provide the user the xmls by default.

I think it can look something like this:
```
<property name="example_field" type="text_line" visableCondition="user.role.id == 1">
      <meta>
           <title lang="nl">Tekst</title>
      </meta>
</property>
```

**Comment by nnatter** [comment]
Think this could be a useful feature and should not be that hard to implement. With 2.2, Sulu provides a `conditionDataProviderRegistry` that allows to inject additional data into the context that is used when evaluating conditions. For example, this is used to make the bundles available via `__bundles`: https://github.com/sulu/sulu/blob/master/src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/conditionDataProviders/bundlesConditionDataProvider.js

I think the following steps would be enough to implement this:

- Include the result of the `getRoles()` method in the serialized user (this is the array that is used by symfony too)
- Register a condition data provider that provide the user that is stored in the `userStore` via `__user`

**Comment by niklasnatter** [comment]
Thanks for reporting! This should be fixed in #6222 and included in the next patch release 🙂

**Comment by jordygroote** [comment]
Great news!
